### PR TITLE
Update build to latest infrastructure

### PR DESCRIPTION
**Pull request by timyates**
- Removed the workaround for secure hashes (fixed in maven central)
- Reinstated the daemon
- Added parallel

Will fail currently as the docs are invalid in places (which is fixed once #509 is merged

### CODE REVIEW

**Comment by melix** [comment]
> Why is spock getting pulled into the junit5 tests and the native image?

https://ge.micronaut.io/s/ofpgo53lbt4n6/dependencies?focusedDependency=WzEsNSwyMjEsWzEsNSxbMjIxXV1d&toggled=W1sxXSxbMSw1XV0

Basically, before your changes, the plugin which was applied to `test-junit5` as `java-library`. Now, it's `io.micronaut.build.internal.module`. This plugin is designed to apply spock automatically: https://github.com/micronaut-projects/micronaut-build/blob/278c63890a5457637a66eaef8d6dfe59addcf291/src/main/groovy/io/micronaut/build/MicronautModulePlugin.groovy#L29

So I think the plugin you want to apply instead is `io.micronaut.build.internal.base-module`.

**Comment by timyates** [comment]
Hmmm, does internal-base-module pull in groovy? https://ge.micronaut.io/s/wy7jtotqntfss/dependencies?dependencies=groovy&expandAll&focusedDependency=WzEsMyw2MTcsWzEsMixbMjFdXV0&focusedDependencyView=versions

**Comment by timyates** [comment]
Seems to...

Adding https://github.com/micronaut-projects/micronaut-test/pull/518/commits/166531b6f96e1db1e64ccb57552719e29a15132a fixes the issue... @melix is this sensible? 🤔

**Comment by melix** [comment]
I am unsure. I would compare the resulting pom/module files when publishing `test-junit5`. If you get Groovy on the classpath and that it wasn't the case before, then it's likely that you get more unwanted dependencies. In which case the base plugin to apply might not be `internal-base-module` but possibly something even more custom based on `io.micronaut.build.internal.base`

**Comment by timyates** [comment]
I'll take a look, thanks for the pointers 👍

File: test-core/build.gradle [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+    id "io.micronaut.build.internal.module"
+}
+
 dependencies {
-    annotationProcessor "io.micronaut:micronaut-inject-java:$micronautVersion"
-    compileOnly "io.micronaut:micronaut-http-server:$micronautVersion"
-    compileOnly("org.spockframework:spock-core:$spockVersion") {
-        exclude group: "org.codehaus.groovy", module: "groovy-all"
-    }
+    annotationProcessor "io.micronaut:micronaut-inject-java"
```
Comment (sdelamo) [inline test-core/build.gradle:6]: I don't think this is necessary.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: test-core/build.gradle (hunk 1 above)
Comment (melix) [reply test-core/build.gradle:6]: Indeed it isn't: https://github.com/micronaut-projects/micronaut-build/blob/278c63890a5457637a66eaef8d6dfe59addcf291/src/main/groovy/io/micronaut/build/MicronautModulePlugin.groovy#L22-L32

Comment (timyates) [reply test-core/build.gradle]: Thanks both, fix pushed: https://github.com/micronaut-projects/micronaut-test/pull/518/commits/22d96a2e9d1fa1e90c575a354e60ecf24ac74701

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
build(17) seems frozen. I am going to merge this.

**Pull request merged**